cmd/containerd-shim-runc-v2/process/exec.go

```diff
@@ -208,11 +208,6 @@ func (e *execProcess) start(ctx context.Context) (err error) {
 		close(e.waitBlock)
 		return e.parent.runtimeError(err, "OCI runtime exec failed")
 	}
-	if e.stdio.Stdin != "" {
```

Comment:
I don't think it's correct. containerd supports exec operation with input. Please add your motivation. Thanks

Reply:
https://github.com/containerd/containerd/blob/6e0dc9f50f4b33d0b1a8c0a3fc64dd67b5b0c920/cmd/containerd-shim-runc-v2/process/exec.go#L240 `openStdin` opens fifo with `O_WRONLY`，which is only used to hold fifo write endpoint. For init process，it is important to prevent init process `sigpipe` when restart containerd, but for exec process, when restart containerd, exec process quit is what we want. Stdin copy for exec process `parent.Platform.CopyConsole` and `pio.Copy` will open stdin fifo with `O_RDONLY` in their own logic.

Reply:
I want to use this patch to solve pty device leak in `CopyConsole` when restart containerd.

Reply:
Synced with @yuqitao offline. I think the right approach is to implement exec operation's GC in CRI level or high-level plugin. The cleanup should be handled async. Since pod exec probe can go crazy, shim could have a lot of running exec processes. It could take time to cleanup. To avoid slow starting, we should handle it async.